x-pack/plugins/apm/server/lib/traces/get_trace.ts

```diff
@@ -55,18 +53,14 @@ export async function getTrace(traceId: string, setup: Setup) {
     }
   };
 
-  interface ServiceBucket {
-    key: string;
-  }
-
   const resp: SearchResponse<Span | Transaction> = await client(
     'search',
     params
   );
 
   return {
-    services: resp.aggregations.services.buckets.map(
-      (bucket: ServiceBucket) => bucket.key
+    services: (resp.aggregations.services.buckets as TermsAggsBucket[]).map(
```

Comment:
From what I can tell, assertion is somewhat frowned upon. https://basarat.gitbooks.io/typescript/docs/types/type-assertion.html#assertion-considered-harmful

I think extending the SearchResponse to include the shape of the aggregations would solve this, but maybe you could also this:
```js
const { buckets }: { buckets: TermsAggsBucket[] } = resp.aggregations.services;
```
?

Reply:
Feel free to just explain why assertion is better here if that's the case, this was the first I'd seen it.